### ISSUE TITLE
Playing a minion into Potion of Polymorph OR a self-transforming minion counts as two cards played for Sherazin

### ISSUE THREAD

**Issue by PattuX** [HearthSim/hs-bugs#727]
* https://www.reddit.com/r/hearthstone/comments/64cgtz/if_you_play_a_minion_into_potion_of_polymorph/

Maybe also bugged with Questing Adventurer? Would be surprising, as there was no report of this.

Self-transformers like Faceless Manipulator also count double: https://youtu.be/Ths1GSopR0Q

**Comment by 10925** [comment]
Probably not.
Since I believe the way Potion of Polymorph coded is to stop current after play phase, and start a new after play phase of sheep.
And Sherazin triggerd at after play phase.

Questing Adventure triggerd at on play phase.

**Comment by PattuX** [comment]
https://youtu.be/24YWlrs5Nj4?t=36s

Also counts towards Hunter Quest.

I wasn't sure on what phase the quests and Sherazin trigger, but your explanation it makes sense.

**Comment by Kybxd** [comment]
I tested with 10925 and found:

1.Dominant Player's Hunter Quest + Secondary Player's Potion of Polymorph:
Quest+1, then Potion triggers, then Quest+1

2.SP's Quest + DP's Potion:
Potion triggers, then Quest+1 twice

I think Potion of Polymorph won't stop the phase, but adds a new After Play Phase for the sheep after it. Secrets and Rumbling Elemental don't trigger on minions left play, but Hunter Quest and Sherazin do. (Dred may also do, but it makes no difference.)

I guess Potion of Polymorph + Rumbling Elemental may also have different results because of the DP bug, but haven't tested yet.

**Comment by Patashu** [comment]
Kybxd found self-transformers also trigger this, so I'll add it here: https://youtu.be/Ths1GSopR0Q

**Comment by Kybxd** [comment]
1.Faceless + Potion count 3 times for Sherazin (https://youtu.be/0rZgPoAUMKs)
2.Facelsss-Battlecry + Rumbling: Mirror doesn't trigger (https://youtu.be/qLcANZCm3bc)
3.Rumbling→KJ→Death→Rumbling (https://youtu.be/igl8rQdB7ps)
4.Rumbling + Potion: Mirror doesn't trigger (https://youtu.be/JxpskFEbtI4)

Conclusion:
1.Faceless's battlecry inserts an After-Play Phase and an After-Summon Phase for the post-transform minion, then the original minion's(Faceless's). And a Death Phase can run between it. This may also apply to all other self-transforming minions.
2.Potion's effect insert an After-Play Phase and an After-Summon Phase for the sheep. It's impossible to verify where are the 2 phases inserted in, for no After-Summon triggers can trigger on minions left play. And  no Death Phase runs until the sequence ends.It should resolves like this: After-Play for the original minion, After-Play for the sheep, After-Summon for the sheep.

**Comment by PattuX** [comment]
Thank for the testing, very interesting! In case you didn't know, we already have a similar mechanic already, the Special Summon Resolution Step (has a section here http://hearthstone.gamepedia.com/Advanced_rulebook#Summon_Resolution_Step), which is basically just a Summon Resolution Step for transformed minions which is inserted at the end of the sequence of playing a minion. Especially the KJ clip suggests that this is very similar in this case.

**Comment by Patashu** [comment]
Yeah but isn't the Summon Resolution Step only for 'whenever you summon an X' triggers, like Steward of Darkshire?

Rumbling Elemental, Knife Juggler, Potion of Polymorph, Sherazin and Mirror Entity are all not SRS triggers, they're either After Play or After Summon triggers.

**Comment by Kybxd** [comment]
One of my friend's further test: (https://youtu.be/q1GGcUU8Cp0)

It seems that the Special SRS is not exactly at the end of the sequence, but for the special phase inserted.

**Comment by Patashu** [comment]
By the way, there's also a Special SRS for minions that transform others with their Battlecry (like Recombobulator and Tinkmaster Overspark). Is there anything special about this?

All of this new info is really interesting, I've added a bunch of TODOs to the advanced rulebook so we don't forget about it, right now I don't know how I want to incorporate it all lol, maybe after it's been tested further and it becomes clearer?

**Comment by PattuX** [comment]
So what I think is that there is an After Play Phase and an After Summon Phase for the transformed minion within the Battlecry Phase. Also I don't think there is any Special SRS, just the normal one.
1. Because the After Play/Summon Phases are within the Battlecry Phase, it makes sense that the SRS triggers resolve after the after play/summon triggers. (https://www.youtube.com/watch?v=GeGVO_n2yA4)
2. The original minion (Faceless/pre-transform Druid of the x) is no longer in play when the outermost After Play/Summon Phase (after the Battlecry Phase) resolves, therefore no secrets or After Summon triggers can trigger. However, Rumbling Elemental triggers (and also quests and Sherazin), even if the minion isn't in play, which is the bug here of course
3. That explains the video kybxd posted: https://www.youtube.com/watch?v=igl8rQdB7ps&feature=youtu.be 
For one, the order is correct (Battlecry phase starts -> After Play for post transform Faceless -> After Summon for post transform Faceless -> End of Battlecry Phase -> Death Phase and SRS -> After Play Phase for pre transform Faceless) and it also explains why there's a Death Phase after Rumbling an KJ triggered.
4. Finally, a log: https://gist.github.com/PattuX/bc02fff2dc288a657cb06df09aa07a2c 
In the first example, the KJ trigger block is within the power (Battlecry) block of the Faceless. Buzzard triggers after that block. Second example is just for comparison, MCT is played, power block is empty, KJ triggers after that.

So basically there are 3 types of sequences for bringing a minion into play then:
1. Playing a minion -> All phases
2. Summoning (and "putting" a minion) a minion -> Only After Summon Phase
3. Transforming a minion -> Only After Play and After Summon Phase

The rules for the two transform phases are basically the same as for summoning.

**Comment by Kybxd** [comment]
Although Rogue Quest is resolved in the same timing, it seems not to apply to this:

1.I have Rogue Quest (Novice Engineer, 3/4) and Faceless a Novice, and the Quest remains 3/4. Similarly, if you have just played Rogue Quest (nothing, 0/4) and Faceless any minion, the Quest will be (Faceless, 1/4).

2.I have Rogue Quest (Novice Engineer, 3/4) and play a Novice, then it's transformed by Potion of Polymorph, the Quest will be 4/4 and then I get the reward.

**Comment by PattuX** [comment]
In this week's Brawl (Servant of Yogg-Saron tryouts), self-transformers seem to cast 2 spells. Happened to me earlier with Shellshifter, but no replay unfortunately.

**Comment by bobbyi** [comment]
Playing a spell that gets Counterspelled does not count as card toward Sherazin. I'd guess this is the flipside to the PoP case; it's counting card-like effects that happened and those don't always match one-to-one to cards played.

**Comment by PattuX** [comment]
That's probably an inteded interaction. Counterspell triggers before the Spell Text Phase ("Whenever you cast a spell"), while Sherazin triggers after the After Spell Phase, just like Wild Pyromancer for example, which will also not trigger, if the spell got countered.

**Comment by bobbyi** [comment]
Sherazin cares about cards being played. Pyro cares about spells being cast. If my spell was countered, I didn't cast a spell, but I did play a card.

**Comment by troggnostupidhs** [comment]
Possibly related to #652

**Comment by troggnostupidhs** [comment]
Potentially fixed

> Sherazin Corpse Flower no longer counts transform minions, such as Shellshifter, twice when dormant.

http://eu.battle.net/hearthstone/en/blog/20811880/hearthstone-update-june-1-01-06-2017

**Comment by Kybxd** [comment]
First I play a 1-cost minion (Raptor Hatchling) into Potion of Polymorph causing Hunter Quest +1.(0/7 to 1/7)
Then I play a non-1-cost minion (Mind Control Tech) into Potion of Polymorph and Hunter Quest doesn't gain +1.
* http://imgur.com/a/Fek3Q

**Comment by VoidMayonnaise** [comment]
https://youtu.be/MPGY-A3Uoxk
Confirmed to be fixed.

**Comment by troggnostupidhs** [comment]
Sherazin Corpse Flower is fixed
